src/init/features/dataconnect/index.ts

```diff
@@ -447,3 +449,20 @@ async function locationChoices(setup: Setup) {
     ];
   }
 }
+
+/**
+ * Converts any string to a DNS friendly service ID.
+ */
+export function toDNSCompatibleId(id: string): string {
+  let defaultServiceId = basename(id)
+    .toLowerCase()
+    .replaceAll(/[^a-z0-9-]/g, "")
+    .slice(0, 61);
```

Comment:
We allow resource names up to 63 characters~

Reply:
Derp, I saw 61 in the regex and turned off my brain. Fixed.